### PR TITLE
LPS-169312 | Test Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# Test Automation for Performance issue with Combine Segments in Dynamic Collections

Hi! :)
This pr is corresponding to this ticket: [LPS-169312](https://issues.liferay.com/browse/LPS-169312)

| **Test File** | **Test Name** |
| --- | --- |
|  Collections.testcase | ViewCombineContentForMultisegmentManualCollectionInAssetPublisher |
|  Collections.testcase | ViewCombineContentForMultisegmentManualCollectionInCollectionDisplayFragment |
|  Collections.testcase | ViewCombineContentForMultisegmentDynamicCollectionInAssetPublisher |
|  Collections.testcase | ViewCombineContentForMultisegmentDynamicCollectionInCollectionDisplayFragment |
|  Collections.testcase | ViewCombineContentOrderForMultisegmentManualCollection |
|  Collections.testcase | ViewCombineContentOrderForMultisegmentDynamicCollection |

### CODE REVIEW

**Comment by rafaellajordao** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase [hunk 1]
```diff
@@ -3731,6 +3731,689 @@ definition {
...
 	}
 
+	@description = "LPS-169312: Given a user belonging to Segment 1 and Segment 2
+	And the configuration to combine assets is enabled
+	When there's a variation of a dynamic collection targeting segment 1 and segment 2, and both are prioritized above the anyone variation
+	And there is an asset present in both variations**
+	Then the user will see the asset only once in Asset Publisher"
+	@priority = "3"
+	test ViewCombineContentForMultisegmentDynamicCollectionInAssetPublisher {
+		property portal.acceptance = "true";
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase:3741]: This test has priority 3 so it is not considered as acceptance. Remove this property.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase [hunk 2]
```diff
@@ -3731,6 +3731,689 @@ definition {
...
+	}
+
+	@description = "LPS-169312: Given a user belonging to Segment 1 and Segment 2
+	And the configuration to combine assets is enabled
+	When there's a variation of a dynamic collection targeting segment 1 and segment 2, and both are prioritized above the anyone variation
+	And there is an asset present in both variations**
+	Then the user will see the asset only once in Collection Display Fragment"
+	@priority = "3"
+	test ViewCombineContentForMultisegmentDynamicCollectionInCollectionDisplayFragment {
+		property portal.acceptance = "true";
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase:3847]: This test has priority 3 so it is not considered as acceptance. Remove this property.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase [hunk 3]
```diff
@@ -3731,6 +3731,689 @@ definition {
...
+	}
+
+	@description = "LPS-169312: Given a user belonging to Segment 1 and Segment 2
+	And the configuration to combine assets is enabled
+	When there's a variation of a manual collection targeting segment 1 and segment 2, and both are prioritized above the anyone variation
+	And there is an asset present in both variations
+	Then the user will see the asset only once in Asset Publisher"
+	@priority = "3"
+	test ViewCombineContentForMultisegmentManualCollectionInAssetPublisher {
+		property portal.acceptance = "true";
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase:3963]: This test has priority 3 so it is not considered as acceptance. Remove this property.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase [hunk 4]
```diff
@@ -3731,6 +3731,689 @@ definition {
...
+	}
+
+	@description = "LPS-169312: Given a user belonging to Segment 1 and Segment 2
+	And the configuration to combine assets is enabled
+	When there's a variation of a manual collection targeting segment 1 and segment 2, and both are prioritized above the anyone variation
+	And there is an asset present in both variations
+	Then the user will see the asset only once in Collection Display Fragment"
+	@priority = "3"
+	test ViewCombineContentForMultisegmentManualCollectionInCollectionDisplayFragment {
+		property portal.acceptance = "true";
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase:4070]: This test has priority 3 so it is not considered as acceptance. Remove this property.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase [hunk 5]
```diff
@@ -3731,6 +3731,689 @@ definition {
...
+	@description = "LPS-169312: Given a user belonging to Segment 1 and Segment 2
+	And the configuration to combine assets is enabled
+	When there's a variation of a dynamic collection targeting segment 1 and segment 2, and both are prioritized above the anyone variation
+	And there are some assets present in both variations**
+	Then the assets shown to the user will be ordered according to:
+	The asset priority (different from the variation priority)
+	The creationDate of the asset"
+	@priority = "3"
+	test ViewCombineContentOrderForMultisegmentDynamicCollection {
+		property portal.acceptance = "true";
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase:4190]: This test has priority 3 so it is not considered as acceptance. Remove this property.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase [hunk 6]
```diff
@@ -3731,6 +3731,689 @@ definition {
...
+	@description = "LPS-169312: Given a user belonging to Segment 1 and Segment 2
+	And the configuration to combine assets is enabled
+	When there's a variation of a manual collection targeting segment 1 and segment 2, and both are prioritized above the anyone variation
+	And there are some assets present in both variations**
+	Then the assets shown to the user will be ordered according to:
+	The priority of the Variation
+	The priority of the asset established in the Variation"
+	@priority = "3"
+	test ViewCombineContentOrderForMultisegmentManualCollection {
+		property portal.acceptance = "true";
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/segmentation/collections/Collections.testcase:4320]: This test has priority 3 so it is not considered as acceptance. Remove this property.

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]

**Comment by rafaellajordao** [comment]
ci:forward

**Pull request closed without merging**